docs/src/components/SearchInput/SearchInput.component.tsx

```diff
@@ -119,18 +139,31 @@ export class SearchInput extends React.PureComponent<InitialState> {
         }
     }
 
+    // Using the label that was selected, uses the associated url and redirects to that page.
+    goTo(value: string) {
+        const result = this.state.dataSource.find((item: DataSet) => {
```

Comment:
I don't think we'll run into any issues with our current layout, but I can imagine this having potential collisions between results with the same name e.g.
```
OVERVIEW
Introduction
CONTRIBUTE
Introduction
```
Clicking the Contribute Introduction would link to the first one. Although, if you do this, I don't think there's any more need for a custom `resultTemplate` (since you have the click behavior defined now).

I think these Autocomplete event handlers (onSelect, etc) should be refactored to pass along the full item information. I also think the label should eventually allow more than just a `string`. But thats outside the scope of this PR.

Reply:
I agree. I was surprised that only the value of `label` was passed, it makes more sense for the payload be the object. I haven't dug into `AutoComplete` yet, but I'm wondering if only the value of the `input` is being passed (which makes sense in a way). For now, since it is using `find` it'll only ever return the first match so this isn't breaking. Worth making a ticket for `AutoComplete` because ultimately I think that's a good enhancement for it to have.

Reply:
I should add that the result template is still necessary because I think allowing the user to click on the result should still be an option. That behavior doesn't exist in `ListItem` otherwise.

Reply:
Click behavior still worked when I tried it (removing `resultTemplate={Results}`) locally.

Reply:
I stand corrected. I didn't realize the `onSelect` prop worked for both selection via hitting `enter` and also for clicking a result. I removed the custom results template.